controllers/sriovnetworknodepolicy_controller.go

```diff
@@ -192,11 +169,23 @@ func (r *SriovNetworkNodePolicyReconciler) SetupWithManager(mgr ctrl.Manager) er
 				Info("Enqueuing sync for delete event", "resource", e.Object.GetName())
 			qHandler(q)
 		},
+		GenericFunc: func(ctx context.Context, e event.GenericEvent, q workqueue.RateLimitingInterface) {
+			log.Log.WithName("SriovNetworkNodePolicy").
+				Info("Enqueuing sync for generic event", "resource", e.Object.GetName())
+			qHandler(q)
+		},
 	}
 
+	// send initial sync event to trigger reconcile when controller is started
+	var eventChan = make(chan event.GenericEvent, 1)
```

Comment:
I think a better solution here will be to move the logic for the nodeState creation into the sriovOperatorConfig controller this way we know the nodeSelector for the config daemons

Reply:
Shouldn't same controller handle the create update and removal of the obj ?

If we go this way, IMO we should handle this in a separate PR IMO.